services/src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutTransactionService.java

```diff
@@ -266,10 +285,8 @@ public PaymentResponse reverseAuthorize(PaymentRequest paymentRequest) {
 
     @Override
     public PaymentResponse refund(PaymentRequest paymentRequest) {
-        PaymentResponse paymentResponse =
-                new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
-                        PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT)
-                                .transactionType(DefaultTransactionTypes.REFUND);
+        PaymentResponse paymentResponse = paymentResponseUtil.buildPaymentResponse(paymentRequest,
+                PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT, DefaultTransactionTypes.REFUND);
```

Comment:
We should move the `paymentResponseUtil.buildPaymentResponse(...)` call inside of the `try` block & we should catch `Exception` instead of `PaymentException`, to guarantee that this method will always return a PaymentResponse